Improved ubuntu build scripts

cyberang3l opened the pull request:
Pull request related to issue #353.
Detailed in information about the pull request in the issue #353, as well as the commit messages.

cyberang3l committed:
Minor improvements for ubuntu build scripts.

* Consistent comments accross the three different ubuntu
  scripts.
* The pip is upgraded after being installed. This removes
  a warning the the user has to upgrade pip.
* The script now finds the number of physical cores and
  compiles the eprosima libraries faster with the
  'make -j<number-of-cores>'.
* The eprosima tar.gz files are removed after the compilation
  of the libraries has finished.

Improvements only for the ubuntu_sim_ros_gazebo.sh script:
* Sourcing .bashrc from within a script doesn't seem to be
  reliable. Sourcing directly the ROS setup.bash files is
  the safest way to ensure that the environment variables
  have been updated.
* Before adding the source lines in the .bashrc file, the
  script is now checking that the lines do not already exist
  in order to avoid adding duplicates.
* The distribution id for an ubuntu-based distro such as
  KDE neon doesn't return ubuntu. This makes the command
  rosdep to return some errors. If the 'rosdep install'
  command returns an error, the build script script now
  reexecutes the command 'rosdept install' with the
  '--os ubuntu:xenial', essentially assuming that the distor
  is ubuntu based. A variable 'unsupported_os' is updated in
  this case, and the user is warned with an error message at
  the end of the script's execution that the script assumed
  that the distro is Ubuntu 16.04 and proceeded with the
  installation. Since the script is also called
  'ubuntu_sim_ros_gazebo.sh', I believe it is fair to assume
  that the default OS supported by this script is ubuntu and
  when a user chooses to execute it, they 'know what they are
  doing'.

Signed-off-by: Vangelis Tasoulas <vangelis@tasoulas.net>

cyberang3l committed:
Added the .swp extension in the git ignore list.

Typically a .swp file is generated everytime you edit
a file with different text editos in Linux-based systems.

Signed-off-by: Vangelis Tasoulas <vangelis@tasoulas.net>

hamishwillee commented on build_scripts/ubuntu_sim_ros_gazebo.sh:
```diff
@@ -114,13 +117,18 @@ rosinstall_generator mavlink | tee -a /tmp/mavros.rosinstall
 wstool merge -t src /tmp/mavros.rosinstall
 wstool update -t src
-rosdep install --from-paths src --ignore-src --rosdistro kinetic -y
+if ! rosdep install --from-paths src --ignore-src --rosdistro kinetic -y; then
```
@TSC21 Can you please review/sanity check the ROS/Gazebo changes ?

cyberang3l committed:
Removed 'wget' from one of the 'apt-get install' lines.

Wget is installed by default on ubuntu, and in any case,
even if we choose to install it this should happen before
its first use in the script. Wget is first used to download
ninja before the corresponding 'apt-get install' line.

Signed-off-by: Vangelis Tasoulas <vangelis@tasoulas.net>

hamishwillee commented on en/setup/dev_env_linux_ubuntu.md:
```diff
@@ -96,14 +98,16 @@ pip install pyulog
 tar -xzf eprosima_fastrtps-1-5-0-linux.tar.gz requiredcomponents
 tar -xzf requiredcomponents/eProsima_FastCDR-1.0.7-Linux.tar.gz
-cd eProsima_FastCDR-1.0.7-Linux; ./configure --libdir=/usr/lib; make; sudo make install
+cpucores=$(( $(lscpu | grep Core.*per.*socket | awk -F: '{print $2}') * $(lscpu | grep Socket\(s\) | awk -F: '{print $2}') ))
```
This is a nice feature for  the scripts, and I wouldn't remove it. However "arguably" it is more complicated than needed for people copy-pasting from docs.

cyberang3l replied:
I agree. Updated this part.

cyberang3l committed:
Updated the file dev_env_linux_ubuntu.md (English only)

to reflect the ubuntu build script changes.

Signed-off-by: Vangelis Tasoulas <vangelis@tasoulas.net>

TSC21 reviewed (approved):
All good to me @hamishwillee
